pipeline.py

```diff
@@ -69,31 +93,121 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+
+        for article in articles:
+            raw_text = article.get_raw_text()
+            processed_tokens = self._process(raw_text)
+
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+
+            for processed_token in processed_tokens:
+                cleaned_tokens.append(processed_token.get_cleaned())
+                single_tagged_tokens.append(processed_token.get_single_tagged())
+                multiple_tagged_tokens.append(processed_token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged_tokens), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged_tokens), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        # Gets rid of '-\n's that break up words, so we won't get 'кот-' 'орый' as example (Should be: 'который')
+        text = raw_text.replace('-\n', '')
+
+        result = Mystem().analyze(text)
+
+        morphological_tokens = []
+
+        morph = pymorphy2.MorphAnalyzer()
+
+        for token_info in result:
+            original_word = token_info['text']
+
+            if not re.match(r'[а-яА-Яa-zA-Z]', original_word):
+                continue
+
+            morphological_token = MorphologicalToken(original_word=original_word)
+
+            if 'analysis' not in token_info:
+                continue
+
+            if not token_info['analysis']:
+                continue
+
+            if not {'lex', 'gr'}.issubset(set(token_info['analysis'][0])):
```

Comment:
very unusual and non-Pythonic solution. what do you want to achieve here?

elegant solutions (especially one-liners) are not always the best if they make explicit and simple things implicit and difficult to understand